File: internal/content/testdata/bench/rust/bounded-increment-sequence-sum.rs
fn bounded_sequence_sum(n: i64, b: i64, x: i64, y: i64) -> i64 {
    let n = n as usize;
    let mut a = vec![0i64; n + 1];
    for i in 1..=n {
        if a[i - 1] + x <= b {
            a[i] = a[i - 1] + x;
        } else {
            a[i] = a[i - 1] - y;
        }
    }
    a.iter().sum()
}

File: internal/content/testdata/bench/csharp/bounded-increment-sequence-sum.cs
public class Solution {
    public long bounded_sequence_sum(long n, long b, long x, long y) {
        long[] a = new long[n + 1];
        a[0] = 0;
        for (int i = 1; i <= n; i++) {
            if (a[i - 1] + x <= b)
                a[i] = a[i - 1] + x;
            else
                a[i] = a[i - 1] - y;
        }
        long sum = 0;
        for (int i = 0; i <= n; i++) sum += a[i];
        return sum;
    }
}
